packages/augur-sdk/src/api/Gnosis.ts

```diff
@@ -305,32 +301,24 @@ export class Gnosis {
         setupData,
       });
 
-      this.safesToCheck.push({
-        safe: ethUtil.toChecksumAddress(response.safe),
-        owner: params.owner,
-        status: GnosisSafeState.WAITING_FOR_FUNDS,
-      });
-
-      this.augur.setGnosisStatus(GnosisSafeState.WAITING_FOR_FUNDS);
-
-      return response;
-    } catch(error) {
-      const restoreAddress = await this.calculateGnosisSafeAddress({
-        payment: '1',
+      const calculatedSafeAddress = await this.calculateGnosisSafeAddress({
+        ...response,
         owner: params.owner,
-        safe: null,
-        paymentToken: null,
       });
+      if(ethUtil.toChecksumAddress(calculatedSafeAddress) !== ethUtil.toChecksumAddress(response.safe)) {
+        this.augur.setGnosisStatus(GnosisSafeState.ERROR);
+        throw new Error(`Potential malicious relay. Returned ${response.safe}. Expected ${calculatedSafeAddress}`);
+      }
 
       this.safesToCheck.push({
-        safe: ethUtil.toChecksumAddress(restoreAddress),
+        safe: ethUtil.toChecksumAddress(response.safe),
         owner: params.owner,
         status: GnosisSafeState.WAITING_FOR_FUNDS,
       });
 
       this.augur.setGnosisStatus(GnosisSafeState.WAITING_FOR_FUNDS);
-      throw ethUtil.toChecksumAddress(restoreAddress);
-    }
+
+      return response;
```

Comment:
I have a mild general concern about returning the `response` here rather than a specifically constructed object where we know exactly its contents.  Following this object around the entire codebase to figure out if there are any attack vectors by controlling its contents is much harder than just checking that the original function does proper validation.